attribute.go

```diff
@@ -24,6 +25,29 @@ type Attribute struct {
 	Data []byte
 }
 
+// IsNested interprets the Type field of the Attribute to determine whether it's
```

Comment:
I think this can be shortened to:

```
// IsNested determines if an Attribute's data contains additional nested attributes.
```

The rest is an implementation detail IMO.